zebra/zebra_nhg.c

```diff
@@ -2210,24 +2257,23 @@ static int nexthop_active(struct nexthop *nexthop, struct nhg_hash_entry *nhe,
 		}
 
 		if (match->type == ZEBRA_ROUTE_CONNECT) {
-			/* Directly point connected route. */
-			newhop = match->nhe->nhg.nexthop;
-			if (newhop) {
-				if (nexthop->type == NEXTHOP_TYPE_IPV4
-				    || nexthop->type == NEXTHOP_TYPE_IPV6)
-					nexthop->ifindex = newhop->ifindex;
-				else if (nexthop->ifindex != newhop->ifindex) {
-					if (IS_ZEBRA_DEBUG_RIB_DETAILED)
-						zlog_debug(
-							"%s: %pNHv given ifindex does not match nexthops ifindex found found: %pNHv",
-							__func__, nexthop,
-							newhop);
-					/*
-					 * NEXTHOP_TYPE_*_IFINDEX but ifindex
-					 * doesn't match what we found.
-					 */
-					return 0;
-				}
+			newhop = zebra_nhg_connected_ifindex(rn, match,
+							     nexthop->ifindex);
+
+			nexthop->ifindex = newhop->ifindex;
```

Comment:
confusing to see this assignment, then another copy of the assignment below, then a test that doesn't look like it can fail. I think we're losing the possibility that 'match' / 'newhop' is the best match, but it does not have the right ifindex, and so we should be taking the failure path below.

Reply:
you are right that should be removed.